### PR TITLE
use a named tuple with coefnames for β in simulate

### PR DESCRIPTION
**Pull request by kleinschmidt**
Self-explanatory.  Means that the vector of samples returned by
`getproperty(::MixedModelBootstrap, :β)` is a vector of named tuples, aka a row
table and can be converted into any other table (e.g. a `DataFrame`) easily.

### CODE REVIEW

**Comment by kleinschmidt** [comment]
I changed `coefnames` to `fixefnames` to handle the rank-deficient (pivoted) case, but the rank deficient case from the tests errors on `parametricbootstrap` anyway so I'm not sure whether it works correctly :)

**Comment by kleinschmidt** [comment]
```julia
rng = MersenneTwister(0)
x = rand(rng, 100)
data = columntable((x = x, x2 = 1.5 .* x, y = rand(rng, 100), z = repeat('A':'T', 5)))
model = fit!(LinearMixedModel(@formula(y ~ x + x2 + (1|z)), data))
pbs = parametricbootstrap(rng, 1000, model)
```

```
DimensionMismatch("B has first dimension 3 but needs 2")

Stacktrace:
 [1] trtrs!(::Char, ::Char, ::Char, ::Array{Float64,2}, ::Array{Float64,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/LinearAlgebra/src/lapack.jl:3405
 [2] ldiv!(::Adjoint{Float64,LowerTriangular{Float64,Array{Float64,2}}}, ::Array{Float64,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/LinearAlgebra/src/triangular.jl:661
 [3] fixef!(::Array{Float64,1}, ::LinearMixedModel{Float64}) at /home/dave/.julia/dev/MixedModels/src/linearmixedmodel.jl:361
 [4] (::MixedModels.var"#95#96"{Float64,Array{Float64,1},Float64,Array{Float64,1},MersenneTwister,Int64,Int64,Array{LinearMixedModel{Float64},1},Array{Array{Float64,1},1},Array{Array{Float64,1},1},ReentrantLock})() at /home/dave/.julia/dev/MixedModels/src/simulate.jl:90
 [5] #3 at ./none:0 [inlined]
 [6] iterate at ./generator.jl:47 [inlined]
 [7] collect(::Base.Generator{ProgressMeter.ProgressWrapper{Base.OneTo{Int64}},MixedModels.var"#3#6"{MixedModels.var"#95#96"{Float64,Array{Float64,1},Float64,Array{Float64,1},MersenneTwister,Int64,Int64,Array{LinearMixedModel{Float64},1},Array{Array{Float64,1},1},Array{Array{Float64,1},1},ReentrantLock}}}) at ./array.jl:622
 [8] macro expansion at /home/dave/.julia/packages/ProgressMeter/g1lse/src/ProgressMeter.jl:724 [inlined]
 [9] #replicate#1(::Bool, ::typeof(replicate), ::MixedModels.var"#95#96"{Float64,Array{Float64,1},Float64,Array{Float64,1},MersenneTwister,Int64,Int64,Array{LinearMixedModel{Float64},1},Array{Array{Float64,1},1},Array{Array{Float64,1},1},ReentrantLock}, ::Int64) at /home/dave/.julia/dev/MixedModels/src/utilities.jl:114
 [10] #replicate at ./none:0 [inlined]
 [11] #parametricbootstrap#94(::Array{Float64,1}, ::Float64, ::Array{Float64,1}, ::Bool, ::typeof(parametricbootstrap), ::MersenneTwister, ::Int64, ::LinearMixedModel{Float64}) at /home/dave/.julia/dev/MixedModels/src/simulate.jl:82
 [12] parametricbootstrap(::MersenneTwister, ::Int64, ::LinearMixedModel{Float64}) at /home/dave/.julia/dev/MixedModels/src/simulate.jl:66
 [13] top-level scope at In[7]:1
```

File: src/simulate.jl [hunk 1]
```diff
@@ -90,7 +92,7 @@ function parametricbootstrap(
         (
          objective = mod.objective,
          σ = mod.σ,
-         β = SVector{p,T}(fixef!(βsc, mod)),
+         β = NamedTuple{β_names}(SVector{p,T}(fixef!(βsc, mod))),
```
[the remark this answers is not in the document]
Comment (palday) [reply src/simulate.jl]: done

**Comment by palday** [comment]
@kleinschmidt Do you want to rebase and force push? Basically, you just need to keep both sets of changes (which do different types of argument checking).

**Comment by kleinschmidt** [comment]
should be good to go now...

**Comment by palday** [comment]
I'm also happy. Just want final confirmation from @dmbates that I haven't introduced any new horrors with my handling of truncating to the rank.

**Pull request merged**